src/multipliers/VotingStreakMultiplier.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {IMultiplier} from "src/interfaces/multipliers/IMultiplier.sol";
+import {YieldDistributor} from "src/YieldDistributor.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+/// @title VotingStreakMultiplier
+/// @notice A contract for managing voting streak multipliers
+/// @dev Implements IMultiplier and IVotingStreakMultiplier interfaces
+contract VotingStreakMultiplier is Initializable, OwnableUpgradeable, IMultiplier {
+    /// @notice The maximum multiplier incrementation
+    uint256 public maxMultiplier;
+
+    /// @notice The increment value for the multiplier
+    uint256 public multiplierIncrement;
+
+    /// @notice The YieldDistributor contract
+    YieldDistributor public yieldDistributor;
+
+    /// @notice Mapping of user addresses to their current multiplier factor
+    mapping(address => uint256) public userToMultiplier;
+
+    /// @notice Mapping of user addresses to their multiplier validity period
+    mapping(address => uint256) public userToValidity;
```

Comment:
I would potentially rename this so it's more clear we're expecting a block number and how the validity is determined. Maybe "userToValidUntil" or "userToExpiresAt".